src/assets/scripts/client/trafficGenerator/SpawnPatternModel.js

```diff
@@ -312,26 +477,106 @@ export default class SpawnPatternModel extends BaseModel {
+     * Return a number to use for the next delay period calculated based
+     * on spawnning method.
      *
      * This is the value that will be used by the `SpawnScheduler` when
      * when creating a new spawn interval.
      *
      * @for SpawnPatternModel
-     * @method getRandomDelayValue
-     * @return randomDelayValue {number}
+     * @method getNextDelayValue
+     * @param gameTime {number}
+     * @return {number}             Next delay period based on spawn method
      */
-    getRandomDelayValue() {
-        // TODO: make this method a fascade by implementing a switch to handle #method variations with other class methods
-        let targetDelayPeriod = this._maximumDelay;
+    getNextDelayValue(gameTime = 0) {
+        switch (this.method) {
+            case SPAWN_METHOD.RANDOM:
+                return this._calculateRandomDelayPeriod();
+            case SPAWN_METHOD.CYCLIC:
+                return this._calculateNextCyclicDelayPeriod(gameTime);
+            case SPAWN_METHOD.SURGE:
+                return this._calculateNextSurgeDelayPeriod(gameTime);
+            case SPAWN_METHOD.WAVE:
+                return this._calculateNextWaveDelayPeriod(gameTime);
+            default:
+                break;
+        }
+    }
 
-        if (targetDelayPeriod < this._minimumDelay) {
-            targetDelayPeriod = this._minimumDelay;
+    /**
+     * Calculates the upper bound of the spawn delay value.
+     *
+     * @for SpawnPatternModel
+     * @method _calculateMaximumDelayFromRate
+     * @return {number}
+     * @private
+     */
+    _calculateMaximumDelayFromRate() {
+        return TIME.ONE_HOUR_IN_SECONDS / this.rate;
+    }
+
+    /**
+     *
+     *
+     * @for SpawnPatternModel
+     * @method _calculateSurgePatternInitialDelayValues
+     * @param spawnPatternJson {object}
+     * @private
+     */
+    _calculateSurgePatternInitialDelayValues(spawnPatternJson) {
+        if (spawnPatternJson.method !== SPAWN_METHOD.SURGE) {
+            return;
         }
 
-        const maxDelayPeriod = targetDelayPeriod + (targetDelayPeriod - this._minimumDelay);
+        // TODO: accept `entrail` param from json
+        this._aircraftPerHourUp = this.speed / this.entrail[0];
+        this._aircraftPerHourDown = this.speed / this.entrail[1];  // to help the uptime calculation
+
+        this.uptime = (this.period * this.rate - this.period * this._aircraftPerHourDown) / (this._aircraftPerHourUp - this._aircraftPerHourDown);
+        this.uptime -= this.uptime % (TIME.ONE_HOUR_IN_SECONDS / this._aircraftPerHourUp);
+        // FIXME: This would better belong in a helper method and should be simplified
+        // adjust to maintain correct acph rate
+        this._aircraftPerHourDown = Math.floor(
+            this.rate * this.period / TIME.ONE_HOUR_IN_SECONDS -
+            Math.round(this._aircraftPerHourUp * this.uptime / TIME.ONE_HOUR_IN_SECONDS)) *
+            TIME.ONE_HOUR_IN_SECONDS / (this.period - this.uptime);
```

Comment:
No flooring / rounding with spawn rates please. And here is the same thing broken up into bitesize pieces if you're interested:
```
const averageSpawnRate = this.rate * this.period * TIME.ONE_SECOND_IN_HOURS;
const elevatedSpawnRate = this._aircraftPerHourUp * this.uptime * TIME.ONE_SECOND_IN_HOURS;
const downTime = this.period - this.uptime;
const hoursSpentAtReducedSpawnRate = downTime * TIME.ONE_SECOND_IN_HOURS;
const reducedSpawnRate = (averageSpawnRate - elevatedSpawnRate) * hoursSpentAtReducedSpawnRate;

this._aircraftPerHourDown = reducedSpawnRate;
```